Automattic/wp-calypso
# Domains: Update domain item list layout

#### Changes proposed in this Pull Request

This PR corresponds to item 3 of pcYYhz-e4-p2, which is a redesign of the domain items list in the site's domains page.

The main updates are:
- Removal of the primary domain card, which is replaced by a "Primary site address" green badge
- The domain item list header was simplified and an "Add a domain to this site" dropdown button was added
- Each domain item component was simplified, with auto-renewal and email information shown as text and the domain connection status shown at the bottom of each card

### Screenshots

This is how the site domains management page looked like:

<img width="1053" alt="Screen Shot 2021-07-22 at 11 29 11" src="https://user-images.githubusercontent.com/5324818/126660739-c2db5f68-56f2-4fd1-bae5-0c207aa2e833.png">

And how it should look like after the update:

<img width="1064" alt="Screen Shot 2021-08-03 at 13 27 53" src="https://user-images.githubusercontent.com/5324818/128052287-19ee4d03-21a3-43ee-bb2a-9194d6ff686c.png">

When the "Add a domain to this site" dropdown button is clicked:

<img width="295" alt="Screen Shot 2021-08-03 at 13 28 24" src="https://user-images.githubusercontent.com/5324818/128052343-1ac8a02f-8cc2-497e-ae60-0ed5360a37f7.png">

Detail of domain item "Verifying connection" status:

<img width="1055" alt="Screen Shot 2021-08-03 at 13 30 24" src="https://user-images.githubusercontent.com/5324818/128052364-c95b31ef-4510-47ac-a0ec-2b052d9ea7a5.png">

And when viewing on a mobile device:

<img width="418" alt="Screen Shot 2021-08-03 at 13 29 02" src="https://user-images.githubusercontent.com/5324818/128052390-e4f992de-d14e-4c82-bc65-222f7dcb9839.png">

#### Testing instructions

The ideal way to test this would be to check the most domain statuses as you can and ensure the layout and info shown are accurate:
- Mapped domains
  - Mapped less than 72 hours ago and still not resolving to WPCOM (should show "Verifying connection" status)
  - Mapped more than 72 hours ago and still not resolving to WPCOM (should show "Connection error" status)
- Registered domains
- Transferred domains

With:
- Configured emails
- Expiring soon (in 30 days or less)
- Any other variations you can think of

Also please check the "Manage all domains" page to ensure information is shown correctly there as well, as they use the same `DomainItem` component.
